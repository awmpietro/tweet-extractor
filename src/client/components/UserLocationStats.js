import React from 'react';
import { Table, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

const UserLocationStats = ({stats}) => {

    const renderUserLocations = () => {
        const user_location = []
        for(let i in stats.user_location){
          user_location.push(
            <tr key={i}>
              <td>{i === "" ? "Undefined" : i}</td>
              <td>{stats.user_location[i]}</td>
            </tr>
          )
        }
        return user_location;
    }

    return(
        <>
        <CardTitle><h4>User Locations</h4></CardTitle>
        <Table bordered striped>
            <thead>
            <tr>
                <th>Location</th>
                <th>Quantity</th>
            </tr>
            </thead>
            <tbody>
            {renderUserLocations()}
            </tbody>
        </Table>
        </>
    )
}

UserLocationStats.propTypes = {
    stats: PropTypes.object
};

export default UserLocationStats;