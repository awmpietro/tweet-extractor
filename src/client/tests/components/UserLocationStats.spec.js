import React from 'react';
import UserLocationStats from '../../components/UserLocationStats';
import renderer from 'react-test-renderer';

describe('UserLocationStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<UserLocationStats stats={{}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});