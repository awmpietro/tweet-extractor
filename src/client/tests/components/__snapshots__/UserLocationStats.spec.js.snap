// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLocationStats snapshot renders 1`] = `
Array [
  <div
    className="card-title"
  >
    <h4>
      User Locations
    </h4>
  </div>,
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Location
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
